docs/src/configuration/app/source-operations.md

```diff
@@ -51,9 +51,39 @@ Note that these operations run in an isolated container which is not part of the
 
 Also, if multiple applications in a single project both result in a new commit, that will appear as two distinct commits in the Git history but only a single new build/deploy cycle will occur. If multiple applications define source operations with the same name, they will all be executed sequentially on each application.
 
+## Source Operations usage examples
+
+### Update a site from an upstream repository or template
+
+This example will synchronize your branch with the content from an upstream Git repository.
+
+You need to add a project-level variable named `env:UPSTREAM_REMOTE` with the Git URI of the upstream repository. That will make that repository available as a Unix environment variable in all environments, including in the Source Operations environment.
+
+Then, define a Source Operation to that upstream repository:
+
+```
+source:
+    operations:
+        upstream-update:
+            command: |
+                set -e
+                git remote add upstream $UPSTREAM_REMOTE
+                git fetch --all
+                git merge upstream/master
+```
+
+Every time the `upstream-update` source operation is ran on a branch, that branch will get checked out, then add a Git remote for the upstream Git repository, then merge the latest changes from its main branch (called `master` in this example). 
+Because there are now local changes committed in the local repository, those changes will then get pushed, built, and deployed. If there’s a merge conflict, the script will fail without committing anything.
```

Comment:
```suggestion
If there’s a conflict merging from the upstream repository, the source operation will fail and not update from upstream.
```

I'm not sure what you mean by the local changes - where do those come from @GuGuss ?

Reply:
What you wrote is correct. Accepted.